etc/scripts/generate-github-test-data.py

```diff
@@ -0,0 +1,140 @@
+# HOW TO USE ME?
+# `python3 /Users/jkay/repos/github-for-jira/etc/scripts/generate-github-test-data.py
+# Generate a personal access token from GitHub(PAT)
+# replace ACCESS_TOKEN with the PAT
+# replace USERNAME with GitHub username
+#
+import requests
+import os
+import subprocess
+import time
+import random
+
+# GitHub API base URL
+BASE_URL = 'https://api.github.com'
+
+# GitHub access token
+ACCESS_TOKEN = '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'
+
+# GitHub username
+USERNAME = 'joshkay10'
+
+# Number of branches and loop iterations
+NUM_BRANCHES = 5
+LOOP_ITERATIONS = 3
+
+# Helper function to make authenticated API requests
+def make_api_request(method, url, headers=None, data=None):
+    headers = headers or {}
+    headers['Authorization'] = f'Bearer {ACCESS_TOKEN}'
+    response = requests.request(method, url, headers=headers, json=data)
+    try:
+        response.raise_for_status()
+        return response.json()
+    except requests.exceptions.HTTPError as e:
+        print(f'Error making API request: {e}')
+        print(f'Response: {response.text}')
+        raise
+
+# Helper function to execute bash commands
+def run_bash_command(command):
+    process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+    return process.returncode, stdout, stderr
+
+# Generate a unique repository name based on epoch time
+repo_name = f'repo-{int(time.time())}'
+
+try:
+    # Step 1: Create a new repository
+    repo_data = {
+        'name': repo_name
+    }
+    create_repo_url = f'{BASE_URL}/user/repos'
+    make_api_request('POST', create_repo_url, data=repo_data)
+    print(f'Repository "{repo_name}" created successfully.')
+
+    # Step 2: Initialize the repository, add README, and push initial commit
+    os.mkdir(repo_name)
+    os.chdir(repo_name)
+    os.system('git init')
```

Comment:
Similar to the suggestion up the top: there's a dedicate library called GitPython. This might actually be worth considering using. From what I've read, it's more efficient, providers better error handling and a bunch of other stuff.

Reply:
Ill look into this, gets a but messy doing it this way after a while